### PR TITLE
[specdec][v1] Fix sampler peak memory utilization when using specdec

### PR DESCRIPTION
**Pull request by Flechman**
When capturing peak memory utilization of the rejection sampler, vLLM v1 assumes a single drafted token. This doesn't match the worst case where all requests draft k>1 tokens, which can lead to OOM.

With `max_num_seq = 200`, num drafted tokens = 5:
- the previous code has logits shape (200+200, vocab_size:=131072) so 400*131072*4 bytes = 209MiB.
- the new code has logits shape (200*5+200, 131072) = 629MiB

### CODE REVIEW

**Review by MatthewBonanni — APPROVED** [review]
LGTM, thanks for the fix!
